### PR TITLE
fix: explicitly activate nf_conntrack on mariner

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Fixes a bug where custom linux os config would not respected for certain sysctls related to the nf_conntrack module on Mariner VHDs

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/pre-install-dependencies.sh [hunk 1]
```diff
@@ -29,6 +29,9 @@ if [[ $OS == $MARINER_OS_NAME ]]; then
   chmod 755 /opt
   chmod 755 /opt/azure
   chmod 644 ${VHD_LOGS_FILEPATH}
+
+  # explicitly activate nf_conntrack so custom sysctls can be properly set 
+  echo nf_conntrack >> /etc/modules-load.d/contrack.conf
```
Comment (hbeberman) [inline vhdbuilder/packer/pre-install-dependencies.sh:34]: I suggest moving this into a function in [tool_installs_mariner.sh](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/scripts/linux/mariner/tool_installs_mariner.sh) then calling it in this block of [install-dependencies.sh](https://github.com/Azure/AgentBaker/blob/4e596e6763a0407a839aff08ea357438e2125aad/vhdbuilder/packer/install-dependencies.sh#L89) to keep as many of the config fixes in one place as possible.

**Review by dmcilvaney — APPROVED** [review]

**Review by alexeldeib — APPROVED** [review]
can we follow up with live VM validation for this? looks like you already started offline

cc @hbeberman

**Comment by alexeldeib** [comment]
oops didn't realize Henry and Daniel already commented :)

File: vhdbuilder/packer/pre-install-dependencies.sh (hunk 1 above)
Comment (cameronmeissner) [reply vhdbuilder/packer/pre-install-dependencies.sh]: > can we follow up with live VM validation for this? looks like you already started offline
> 
> cc @hbeberman

already working on the PR for that :)

**Pull request merged**